prometheus rerun query on input update

fxmiii opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our [`CONTRIBUTING.md`](https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md) guide.
2. Ensure you have added or ran the appropriate tests for your PR.
3. If it's a new feature or config option it will need a docs update. Docs are under the docs folder in repo root.
4. If the PR is unfinished, mark it as a draft PR.
5. Rebase your PR if it gets out of sync with master
6. Name your PR as `<FeatureArea>: Describe your change`. If it's a fix or feature relevant for changelog describe the user  impact in the title. The PR title is used in changelog for issues marked with `add to changelog` label. 
-->

**What this PR does / why we need it**: When a user updates the fields for "Legend" and "Min step" the query is not re-run.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

**Special notes for your reviewer**: I think this functionality at some point moved from '/public/app/plugins/datasource/prometheus/partials/query.editor.html' where this used to work by calling "ctrl.refreshMetricData()". Maybe there is a good reason not to do it anymore but I can't figure out why.

Also I fixed a small typo in a comment in the file. Maybe I should have done that in a separate PR?

Thanks

fxmiii committed:
Rerun query on input changes

Added "this.onRunQuery" for update of the input fields so that the panel updates when changed

fxmiii committed:
typo in comment

changed "fullu" to "fully"

davkal reviewed (changes requested):
The input change handlers should not trigger expensive actions like queries, since they are part of the tight update loop of react components. 
I recommend registering an `onBlur` handler on both inputs, and have that trigger the query run. AFAIK that was the original behavior (run query when leaving the field). 
Let me know if you need help doing this. @fxmiii

fxmiii committed:
modified to onBlur for legend, interval

legend and interval (min step) now call for query re-run when they go 'onBlur' instead of 'onChange'

fxmiii commented:
Is this what you are looking for?

fxmiii commented:
I suppose it could also be smarter by only rerunning the query onBlur if the field has changed. But that also might be unnecessary complexity for little real benefit.

davkal committed:
PromQueryEditor: Updated snapshot
